mu/logic.py

```diff
@@ -525,8 +525,8 @@ def save(self):
             tab.path = self._view.get_save_path(workspace)
         if tab.path:
             # The user specified a path to a file.
-            if not os.path.basename(tab.path).endswith('.py'):
-                # No extension given, default to .py
+            if os.path.splitext(os.path.basename(tab.path))[1] == '':
```

Comment:
Call to `os.path.basename()` is unnecessary as `os.path.splitext` handles it internally

```py3
>>> import os
>>> os.path.splitext('/non.sense/to.test/this/thing.py.in')
('/non.sense/to.test/this/thing.py', '.in')
>>> os.path.splitext('/non.sense/to.test/this/thing')
('/non.sense/to.test/this/thing', '')
>>> 
```